packages/@lwc/engine-dom/src/stylesheets.ts

```diff
@@ -0,0 +1,41 @@
+import { create, isUndefined } from '@lwc/shared';
+
+const elementsToStyleContents = new WeakMap<Element, { [content: string]: true } | undefined>();
+const elementsToStyleElements = new WeakMap<Element, HTMLStyleElement | undefined>();
```

Comment:
These `WeakMap`s are kind of ugly, but nothing better popped into my head as a way to keep track of which styles (strings) had been added at a given mount point, and which style elements had already been inserted for those mount points.